core/src/main/java/com/adyen/checkout/core/mbway/internal/ui/model/MBWayTransformerRegistry.kt

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2025 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by ararat on 24/1/2025.
+ */
+
+package com.adyen.checkout.core.mbway.internal.ui.model
+
+import com.adyen.checkout.core.internal.ui.state.transformer.DefaultTransformer
+import com.adyen.checkout.core.internal.ui.state.transformer.FieldTransformer
+import com.adyen.checkout.core.internal.ui.state.transformer.FieldTransformerRegistry
+import com.adyen.checkout.core.mbway.internal.ui.state.MBWayFieldId
+
+internal class MBWayTransformerRegistry : FieldTransformerRegistry<MBWayFieldId> {
+    private val transformers = MBWayFieldId.entries.associateWith { fieldId ->
+        when (fieldId) {
+            MBWayFieldId.COUNTRY_CODE -> DefaultTransformer()
+            MBWayFieldId.PHONE_NUMBER -> LocalPhoneNumberTransformer()
+        }
+    }
+
+    @Suppress("UNCHECKED_CAST")
+    override fun <T> transform(key: MBWayFieldId, value: T): T {
```

Comment:
In the interface the field is called `fieldId`, so it's better to use it here as well.